### PR TITLE
Update with many small changes/fixes

### PR DESCRIPTION
**Pull request by HomerW**
This PR adds many small changes/fixes we've made internally. 
- Variable naming changes
- Dataloader now returns actions that are already chunked
- Creating the proipio key is now handled by standardization functions
- New UNet diffusion head
- Action unnormalization happens in the model now
- Updated visualizers
- Fixed bug in attention mask, added `use_correct_attention` flag to maintain backwards compatibility
- Added `supply_rng` function wrapper that can be used to wrap the model's `sample_actions` function to ensure that a fresh random key is used for each call
- Fixed issue where image augmentation wasn't being applied because in configs `image_augment_kwargs` was not a dict

### CODE REVIEW

File: examples/03_eval_finetuned.py [hunk 1]
```diff
@@ -23,12 +24,15 @@
 import numpy as np
 import wandb
 
-sys.path.append("path/to/your/act")
+# sys.path.append("path/to/your/act")
+sys.path.append("/nfs/nfs2/users/homer/act")
```
Comment (dibyaghosh) [inline examples/03_eval_finetuned.py:28]: Remove this

File: examples/03_eval_finetuned.py [hunk 2]
```diff
@@ -67,9 +69,11 @@ def main(_):
     env = HistoryWrapper(env, horizon=1)
     env = RHCWrapper(env, exec_horizon=50)
 
-    # wrap env to handle action/proprio normalization -- match normalization type to the one used during finetuning
-    env = UnnormalizeActionProprio(
-        env, model.dataset_statistics, normalization_type="normal"
+    policy_fn = supply_rng(
```
Comment (dibyaghosh) [inline examples/03_eval_finetuned.py:72]: Can you add a small comment here that explains that the `supply_rng` helper creates a fresh rng every time it's called

File: octo/data/oxe/oxe_dataset_mixes.py [hunk 3]
```diff
@@ -10,6 +10,7 @@
     ("kuka", 0.8341046294),
     ("bridge_dataset", 1.0),
     ("taco_play", 2.0),
+    ("taco_extra", 2.0),
```
Comment (dibyaghosh) [inline octo/data/oxe/oxe_dataset_mixes.py:13]: Is this dataset publicly available?

File: octo/data/utils/data_utils.py [hunk 4]
```diff
@@ -181,71 +192,16 @@ def get_dataset_statistics(
     return metadata
 
 
-def combine_dataset_statistics(
```
Comment (dibyaghosh) [inline octo/data/utils/data_utils.py:184]: Why did we delete this function? is this what Sid submitted as an external PR?

**Review by dibyaghosh — APPROVED** [review]
Mainly checking that the examples are still consistent

**Review by kpertsch — APPROVED** [review]
I didn't verify whether all the proprio computations make sense for the OpenX datasets -- did you use the info from the former configs to figure this out?

File: octo/data/traj_transforms.py [hunk 5]
```diff
@@ -87,6 +108,18 @@ def subsample(traj: dict, subsample_length: int) -> dict:
     return traj
 
 
+def zero_out_future_proprio(traj: dict) -> dict:
```
[the remark this answers is not in the document]
Comment (HomerW) [reply octo/data/traj_transforms.py:111]: Ah yeah, makes sense. Though it looks like this function is not used anywhere right now. Should we just delete it instead?

File: examples/04_eval_finetuned_on_robot.py [hunk 6]
```diff
@@ -129,22 +122,19 @@ def sample_actions(
...
-            nonlocal rng
-            rng, key = jax.random.split(rng)
-            return f(*args, rng=key, **kwargs)
-
-        return wrapped
-
-    policy_fn = supply_rng(
-        partial(
+    policy_fn = partial(
+        supply_rng(
```
Comment (dibyaghosh) [inline examples/04_eval_finetuned_on_robot.py:133]: Something messed up is happening here

Comment (HomerW) [reply examples/04_eval_finetuned_on_robot.py]: fixed

File: octo/model/components/tokenizers.py [hunk 7]
```diff
@@ -202,7 +203,7 @@ def __call__(
             assert self.proper_pad_mask, "Cannot skip unless using proper pad mask."
             return None
 
-        if not isinstance(tasks["language_instruction"], jax.Array):
+        if not isinstance(tasks["language_instruction"], (jax.Array, np.ndarray)):
```
Comment (dibyaghosh) [inline octo/model/components/tokenizers.py:206]: When is this ever an np array?

File: octo/model/octo_model.py [hunk 8]
```diff
@@ -175,29 +187,46 @@ def sample_actions(
...
             train=train,
             argmax=argmax,
             sample_shape=sample_shape,
             rng=rng,
             temperature=temperature,
+            embodiment_action_dim=len(unnormalization_statistics["mean"])
+            if unnormalization_statistics is not None
+            else None,
         )
+        if unnormalization_statistics is not None:
```
Comment (dibyaghosh) [inline octo/model/octo_model.py:218]: Did we completely drop support for bounds normalization? Could there be an issue here where they train on bounds normalization, but accidentally evaluate with normal unnormalization

Comment (HomerW) [reply octo/model/octo_model.py:221]: Good catch, I updated sample_actions to take in a normalization type.

File: octo/model/components/tokenizers.py (hunk 7 above)
Comment (HomerW) [reply octo/model/components/tokenizers.py:206]: @SudeepDasari I think this got added in the new action head PR?

**Comment by mees** [comment]
LGTM!
